pkg/client/unit_test.go

```diff
@@ -79,8 +79,7 @@ func TestUnitUpdateState(t *testing.T) {
 }
 
 func TestUnitUpdateAPMConfig(t *testing.T) {
-
-	var initialUnitNoAPM = Unit{
+	initialUnitNoAPM := &Unit{
```

Comment:
This struct is now shared among different testcases which makes the test fail when all the testcases are run. Just tried to nest it at the cost of some duplication but that take care of the failures without reintroducing the `go vet` violations

Reply:
Added a new commit where we create a new unit per testcase

Reply:
I was pushing a fix as well, just copying and pasting it all over. Yours is more elegant. Thanks